src/paradigma/heart_rate/tfd.py

```diff
@@ -0,0 +1,553 @@
+import numpy as np
+from scipy import signal
+
+"""
+
+This module contains the implementation of the Generalized Time-Frequency Distribution (TFD) computation using non-separable kernels.
+This is a Python implementation of the MATLAB code provided by John O Toole in the following repository: https://github.com/otoolej/memeff_TFDs 
+
+The following functions are implemented for the computation of the TFD:
+    - nonsep_gdtfd: Computes the generalized time-frequency distribution using a non-separable kernel.
+    - get_analytic_signal: Generates the analytic signal of the input signal.
+    - gen_analytic: Generates the analytic signal by zero-padding and performing FFT.
+    - gen_time_lag: Generates the time-lag distribution of the analytic signal.
+    - multiply_kernel_signal: Multiplies the TFD by the Doppler-lag kernel.
+    - gen_doppler_lag_kern: Generates the Doppler-lag kernel based on kernel type and parameters.
+    - get_kern: Gets the kernel based on the provided kernel type.
+    - get_window: General function to calculate a window function.
+    - get_win: Helper function to create the specified window type.
+    - shift_window: Shifts the window so that positive indices appear first.
+    - pad_window: Zero-pads the window to a specified length.
+    - compute_tfd: Finalizes the time-frequency distribution computation.
+
+
+"""	
+
+def nonsep_gdtfd(x, kern_type=None, kern_params=None):
```

Comment:
We have used `Union` so far, but apparently the correct way since Python 3.10 is using Pipe: 

```suggestion
def nonsep_gdtfd(
x: np.ndarray, 
kern_type: None | str = None, 
kern_params: None | str = None
):
```